sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -124,29 +124,25 @@ public void setSideInputs(Map<PCollectionView<?>, Iterable<WindowedValue<?>>> si
    * <p>If this isn't called, {@code DoFnTester} assumes the
    * {@code DoFn} takes no side inputs.
    */
-  public void setSideInput(PCollectionView<?> sideInput, Iterable<WindowedValue<?>> value) {
-    sideInputs.put(sideInput, value);
+  public <T> void setSideInput(PCollectionView<T> sideInput, Iterable<WindowedValue<T>> value) {
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    Iterable values = (Iterable) value;
+    sideInputs.put(sideInput, values);
   }
 
   /**
    * Registers the values for a side input {@link PCollectionView} to
    * pass to the {@link DoFn} under test. All values are placed
    * in the global window.
    */
-  public void setSideInputInGlobalWindow(
-      PCollectionView<?> sideInput,
-      Iterable<?> value) {
-    sideInputs.put(
-        sideInput,
-        Iterables.transform(value, new Function<Object, WindowedValue<?>>() {
-          @Override
-          public WindowedValue<?> apply(Object input) {
-            return WindowedValue.valueInGlobalWindow(input);
-          }
-        }));
+  public <T> void setSideInput(
```

Comment:
Why this change? The additional method for just a single value seems fine, but the case for adding globally windowed data remains. At the least, update the javadoc to indicate that you aren't setting the entirety of the PCollection contents. Is it unused?